From 2fba5d338590e8cce6c0a0de04feeb48d7a470cb Mon Sep 17 00:00:00 2001
Message-Id: <2fba5d338590e8cce6c0a0de04feeb48d7a470cb.1441918786.git.yunwei.zhang@intel.com>
In-Reply-To: <8631704be06e00cb7a7a415f13811ca4ece99e8b.1441918786.git.yunwei.zhang@intel.com>
References: <8631704be06e00cb7a7a415f13811ca4ece99e8b.1441918786.git.yunwei.zhang@intel.com>
From: Shubhangi Shrivastava <shubhangi.shrivastava@intel.com>
Date: Thu, 10 Sep 2015 12:10:03 +0530
Subject: [PATCH 02/10] REVERTME [VPG]: drm/i915: Remove unused
 support_dp_audio variable

This patch removes support_dp_audio variable which will not be used
anymore. Audio will be dynamically switched betweet DP and HDMI
with the new design and this has became redundant.

REVERTME: LP audio is not available upstream and may not be for
some more time which is required for this.

Change-Id: I91a816036e7627babbf8febf4c24bac576109575
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-10172
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Signed-off-by: Shubhangi Shrivastava <shubhangi.shrivastava@intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h |  1 -
 drivers/gpu/drm/i915/intel_dp.c | 41 +++++++++++++++++++----------------------
 2 files changed, 19 insertions(+), 23 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index 4d5ad34..97417e6 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -1785,7 +1785,6 @@ struct drm_i915_private {
 	int link_rate;
 	int hdmi_audio_interrupt_mask;
 	struct work_struct hdmi_audio_wq;
-	bool support_dp_audio;
 #endif
 
 	/* Atomicity fixes */
diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index b846dd9..7d4e025 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -4648,7 +4648,7 @@ intel_dp_detect(struct drm_connector *connector, bool force)
 out:
 #ifdef CONFIG_SUPPORT_LPDMA_HDMI_AUDIO
 	if (IS_VALLEYVIEW(dev) && !is_edp(intel_dp) &&
-		dev_priv->support_dp_audio && status != i915_hdmi_state) {
+		status != i915_hdmi_state) {
 
 		/*
 		 * If HDMI status is conencted, the event to audio
@@ -5412,7 +5412,6 @@ intel_dp_init(struct drm_device *dev, int output_reg, enum port port)
 	struct intel_connector *intel_connector;
 #ifdef CONFIG_SUPPORT_LPDMA_HDMI_AUDIO
 	struct hdmi_audio_priv *hdmi_priv;
-	struct drm_i915_private *dev_priv = dev->dev_private;
 #endif
 
 	intel_dig_port = kzalloc(sizeof(*intel_dig_port), GFP_KERNEL);
@@ -5469,28 +5468,26 @@ intel_dp_init(struct drm_device *dev, int output_reg, enum port port)
 	intel_encoder->hot_plug = intel_dp_hot_plug;
 
 #ifdef CONFIG_SUPPORT_LPDMA_HDMI_AUDIO
-	if (dev_priv->support_dp_audio) {
-		hdmi_priv = kzalloc(sizeof(struct hdmi_audio_priv),
-					GFP_KERNEL);
-		if (!hdmi_priv) {
-			pr_err("failed to allocate memory");
-			goto mem_err;
-		}
-		hdmi_priv->dev = dev;
-
-		if (IS_CHERRYVIEW(dev)) {
-			chv_set_lpe_audio_reg_pipe(dev, intel_encoder,
-						hdmi_priv, port);
-		} else {
-			hdmi_priv->hdmi_lpe_audio_reg =
-				I915_HDMI_AUDIO_LPE_B_CONFIG;
-		}
+	hdmi_priv = kzalloc(sizeof(struct hdmi_audio_priv),
+				GFP_KERNEL);
+	if (!hdmi_priv) {
+		pr_err("failed to allocate memory");
+		goto mem_err;
+	}
+	hdmi_priv->dev = dev;
 
-		/* HACK */
-		hdmi_priv->monitor_type = MONITOR_TYPE_HDMI;
-		hdmi_priv->is_hdcp_supported = false;
-		i915_hdmi_audio_init(hdmi_priv);
+	if (IS_CHERRYVIEW(dev)) {
+		chv_set_lpe_audio_reg_pipe(dev, intel_encoder,
+					hdmi_priv, port);
+	} else {
+		hdmi_priv->hdmi_lpe_audio_reg =
+			I915_HDMI_AUDIO_LPE_B_CONFIG;
 	}
+
+	/* HACK */
+	hdmi_priv->monitor_type = MONITOR_TYPE_HDMI;
+	hdmi_priv->is_hdcp_supported = false;
+	i915_hdmi_audio_init(hdmi_priv);
 #endif
 
 mem_err:
-- 
1.9.1

